corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -178,18 +178,18 @@ def send_to_elasticsearch(index_info, doc_type, doc_id, es_getter, name, data=No
     propagate_failure = settings.UNIT_TESTING
     while current_tries < retries:
         try:
-            doc_exists = es_interface.doc_exists(alias, doc_id, doc_type)
             if delete:
-                if doc_exists:
+                try:
                     es_interface.delete_doc(alias, doc_type, doc_id)
-            elif doc_exists:
+                except NotFoundError:
+                    pass
+            else:
                 params = {'retry_on_conflict': 2}
                 if es_merge_update:
                     es_interface.update_doc_fields(alias, doc_type, doc_id, fields=data, params=params)
```

Comment:
you'll also need to catch an exception here since this will return a 404 if the doc doesn't exist. A unit test for that would be good.

Reply:
I have added tests for the docs-missing case, but it looks like there is already a catch block for NotFoundError [here](https://github.com/dimagi/commcare-hq/pull/28609/files#diff-4e69302c53e47b78b5bc53e7d19d7c1eR221), so I have removed the redundant catch in delete as well.